daffodil-test/src/test/scala/org/apache/daffodil/unparser/TestOVCAndLength.scala

```diff
@@ -19,14 +19,10 @@ package org.apache.daffodil.unparser
 
 import org.junit.Test
 import org.junit.AfterClass
-import org.apache.daffodil.util._
-import org.apache.daffodil.tdml.DFDLTestSuite
+import org.apache.daffodil.tdml.Runner
 
 object TestOVCAndLength {
-  val testDir = "/org/apache/daffodil/unparser/"
-  val aa = testDir + "OVCAndLengthTest.tdml"
-  var runner = new DFDLTestSuite(Misc.getRequiredResource(aa))
-
+  var runner = Runner("/org/apache/daffodil/unparser/OVCAndLengthTest.tdml")
   @AfterClass def tearDown(): Unit = {
     runner = null
```

Comment:
should be runner.reset

Reply:
Raises a minor point. Should all these be var's and we set them to null after the reset? Each Runner instance itself doesn't take up much memory once it's reset. It's really just the constructor arguments and and the memory for the Runner itself, so maybe it's not worth it? But I guess this is technically a leak since these Runner's can never be cleaned up? Maybe we want runner.reset to return null so that you could do something like:

```scala
@AfterClass def tearDown(): Unit = {
    runner = runner.reset
}
```
So we call reset to free up cached stuff and delete any tempoary files, and then set the runner to null the the runner instance itself can be free?

Though, what happens if you re-run the test? Isn't the runner null? Or doe var runner somehow get reinitialized? How do we not get a null pointer exception when this test is rerun?

Reply:
Not worth it. Making runner reset return null like that is a fine change, but we're not worried about regression testing Daffodil running TDML tests in a loop indefinitely. That's not an application scenario we care about. We have a few thousand tests, suppose that grows to 100K tests, and there's average 10 tests per runner, so 10K runners. If each is a 10K object that's only 10Mbytes.